### PR TITLE
JCL-251: Increase test coverage

### PR DESCRIPTION
**Pull request by hellikopter**
Changes in PR:
- expand `maven-surefire-plugin` executions to cover all provided implementations

Implementations

| Interface | Implementation | Module |
|-----------|-------------------|----------|
| ClientCache | CaffeineCache, GuavaCache |  caffeine, guava |
| HttpService | HttpClientService, OkHttpService | httpclient, okhhtp |
| JsonService | JacksonService, JsonbService | jackson, jsonb |
| RdfService | JenaService, RDFLegacyService, RDF4JService | jena, rdf-legacy, rdf4j |

Test coverage

| Module | Dependency implementation matrix |
|----------|-----------------------------------------|
| access-grant | [httpclient, okhhtp] x [jackson, jsonb*] x [caffeine, guava] |
| core | [httpclient, okhhtp] x [jackson, jsonb*] x [jena, rdf-legacy*, rdf4j*] |
| integration | [httpclient*, okhhtp] x [jackson, jsonb*] x [jena, rdf-legacy*, rdf4j*] |
| jena | [httpclient, okhhtp] |
| openid | [httpclient, okhhtp] x [jackson, jsonb*] x [caffeine, guava] |
| rdf-legacy | [httpclient, okhhtp] |
| rdf4j | [httpclient, okhhtp] |
| solid | [httpclient, okhhtp*] x [jena, rdf-legacy*, rdf4j] |
| uma | [httpclient, okhhtp] x [jackson, jsonb*] x [caffeine, guava] |
| webid | [httpclient, okhhtp] x [jena, rdf-legacy*, rdf4j] |

Test variations containing module marked with `*` are ~~disabled~~ removed due to the test failures.

### CODE REVIEW

File: access-grant/pom.xml [hunk 1]
```diff
@@ -98,9 +122,124 @@
...
         <artifactId>maven-surefire-plugin</artifactId>
-        <configuration>
-          <systemPropertyVariables />
-        </configuration>
+        <executions>
+          <execution>
+            <id>default-test</id>
+            <phase>none</phase>
+          </execution>
+          <!--<execution>
```
Comment (acoburn) [inline access-grant/pom.xml:130]: This commented-out section can be removed

File: access-grant/pom.xml [hunk 2]
```diff
@@ -98,9 +122,124 @@
...
+            </goals>
+            <configuration>
+              <classpathDependencyExcludes>
+                <classpathDependencyExclude>com.inrupt.client:inrupt-client-okhttp</classpathDependencyExclude>
+                <classpathDependencyExclude>com.inrupt.client:inrupt-client-jsonb</classpathDependencyExclude>
+                <classpathDependencyExclude>com.inrupt.client:inrupt-client-caffeine</classpathDependencyExclude>
+              </classpathDependencyExcludes>
+            </configuration>
+          </execution>
+          <!--<execution>
```
Comment (acoburn) [inline access-grant/pom.xml:186]: Remove

File: core/pom.xml [hunk 3]
```diff
@@ -141,10 +171,192 @@
...
+            <configuration>
+              <classpathDependencyExcludes>
+                <classpathDependencyExclude>com.inrupt.client:inrupt-client-okhttp</classpathDependencyExclude>
+                <classpathDependencyExclude>com.inrupt.client:inrupt-client-jsonb</classpathDependencyExclude>
+                <classpathDependencyExclude>com.inrupt.client:inrupt-client-rdf4j</classpathDependencyExclude>
+                <classpathDependencyExclude>com.inrupt.client:inrupt-client-rdf-legacy</classpathDependencyExclude>
+              </classpathDependencyExcludes>
+            </configuration>
+          </execution>
+          <!--<execution>
```
Comment (acoburn) [inline core/pom.xml:239]: Remove

**Review by acoburn — APPROVED** [review]
This looks really good -- can you remove the commented-out maven config? (I've noted a few locations)

File: core/pom.xml (hunk 3 above)
Comment (hellikopter) [reply core/pom.xml]: Done in [d90f442](https://github.com/inrupt/solid-client-java/pull/483/commits/d90f442745c35dfe21870f5281b3accafe45c966)

File: core/pom.xml [hunk 4]
```diff
@@ -141,10 +171,192 @@
...
-        <configuration>
-          <systemPropertyVariables>
-          </systemPropertyVariables>
-        </configuration>
+        <executions>
+          <execution>
+            <id>default-test</id>
+            <phase>none</phase>
+          </execution>
+          <!--<execution>
```
[the remark this answers is not in the document]
Comment (hellikopter) [reply core/pom.xml]: Done in [d90f442](https://github.com/inrupt/solid-client-java/pull/483/commits/d90f442745c35dfe21870f5281b3accafe45c966)

File: access-grant/pom.xml (hunk 2 above)
Comment (hellikopter) [reply access-grant/pom.xml]: Done in [d90f442](https://github.com/inrupt/solid-client-java/pull/483/commits/d90f442745c35dfe21870f5281b3accafe45c966)

File: access-grant/pom.xml (hunk 1 above)
Comment (hellikopter) [reply access-grant/pom.xml]: Done in [d90f442](https://github.com/inrupt/solid-client-java/pull/483/commits/d90f442745c35dfe21870f5281b3accafe45c966)

**Comment by langsamu** [comment]
> This looks really good -- can you remove the commented-out maven config? (I've noted a few locations)

The commented out `execution`s indicate problems with the code. When we don't have those executions then we're not covering some service implementation constellation. They were commented because they were failing.

Can we do something to remember this @hellikopter, @hellikopter? `TODO`s or tickets?

**Pull request merged**